src/adlist.c

```diff
@@ -93,17 +93,23 @@ list *listAddNodeHead(list *list, void *value)
     if ((node = zmalloc(sizeof(*node))) == NULL)
         return NULL;
     node->value = value;
+    listAddExistingNodeHead(list, node);
+}
+
+/*
+ * Add a node that has already been mallocced to the head of list
+*/
+void listAddExistingNodeHead(list* list, listNode *node) {
     if (list->len == 0) {
-        list->head = list->tail = node;
-        node->prev = node->next = NULL;
-    } else {
-        node->prev = NULL;
-        node->next = list->head;
-        list->head->prev = node;
-        list->head = node;
-    }
-    list->len++;
-    return list;
+            list->head = list->tail = node;
+            node->prev = node->next = NULL;
+        } else {
+            node->prev = NULL;
+            node->next = list->head;
+            list->head->prev = node;
+            list->head = node;
+        }
+        list->len++;
```

Comment:
This looks extra indented
```suggestion
        list->head = list->tail = node;
        node->prev = node->next = NULL;
    } else {
        node->prev = NULL;
        node->next = list->head;
        list->head->prev = node;
        list->head = node;
    }
    list->len++;
```

Reply:
good catch will fix